lib/components/map/default-map.js

```diff
@@ -1,83 +1,152 @@
+import BaseMap from '@opentripplanner/base-map'
+import FromToLocationPicker from '@opentripplanner/from-to-location-picker'
 import React, { Component } from 'react'
 import { connect } from 'react-redux'
+import styled from 'styled-components'
 
 import {
   bikeRentalQuery,
   carRentalQuery,
   vehicleRentalQuery
 } from '../../actions/api'
-import BaseMap from './base-map'
-import EndpointsOverlay from './endpoints-overlay'
-import ParkAndRideOverlay from './park-and-ride-overlay'
-import StopsOverlay from './stops-overlay'
-import StopViewerOverlay from './stop-viewer-overlay'
+import {
+  setLocation,
+  setMapPopupLocation,
+  setMapPopupLocationAndGeocode
+} from '../../actions/map'
+import BoundsUpdatingOverlay from './bounds-updating-overlay'
+import EndpointsOverlay from './connected-endpoints-overlay'
+import ParkAndRideOverlay from './connected-park-and-ride-overlay'
+import RouteViewerOverlay from './connected-route-viewer-overlay'
+import StopViewerOverlay from './connected-stop-viewer-overlay'
+import StopsOverlay from './connected-stops-overlay'
+import TransitiveOverlay from './connected-transitive-overlay'
+import TripViewerOverlay from './connected-trip-viewer-overlay'
+import VehicleRentalOverlay from './connected-vehicle-rental-overlay'
 import TileOverlay from './tile-overlay'
-import TransitiveOverlay from './transitive-overlay'
-import TripViewerOverlay from './trip-viewer-overlay'
-import RouteViewerOverlay from './route-viewer-overlay'
-import VehicleRentalOverlay from './vehicle-rental-overlay'
 import ZipcarOverlay from './zipcar-overlay'
 
+const MapContainer = styled.div`
+  height: 100%;
+  width: 100%;
+
+  .map {
+    height: 100%;
+    width: 100%;
+  }
+
+  * {
+    box-sizing: unset;
+  }
+`
+
 class DefaultMap extends Component {
+  onMapClick = (e) => {
+    this.props.setMapPopupLocationAndGeocode(e)
+  }
+
+  onPopupClosed = () => {
+    this.props.setMapPopupLocation({ location: null })
+  }
+
+  onSetLocationFromPopup = (payload) => {
+    const { setLocation, setMapPopupLocation } = this.props
+    setMapPopupLocation({ location: null })
+    setLocation(payload)
+  }
+
   render () {
     const {
       bikeRentalQuery,
       bikeRentalStations,
       carRentalQuery,
       carRentalStations,
       mapConfig,
+      mapPopupLocation,
       vehicleRentalQuery,
       vehicleRentalStations
     } = this.props
+
+    const center = mapConfig && mapConfig.initLat && mapConfig.initLon
+      ? [mapConfig.initLat, mapConfig.initLon]
+      : null
+
+    const popup = mapPopupLocation && {
+      contents: (
+        <div style={{ width: 240 }}>
+          <div style={{ fontSize: 14, marginBottom: 6 }}>
```

Comment:
With our use of styled-components, please move the styles to a separate variable.